tutorials/tutorial-howto-monitor-django-prometheus.md

```diff
@@ -0,0 +1,230 @@
+exporting prometheus-style monitoring metrics from our Django application.
+
+#### Install django-prometheus
+```bash
+python -m pip install django-prometheus
+```
+
+#### Modify settings.py
+In settings.py, add:
+
+```
+INSTALLED_APPS = [
+   ...
+   'django_prometheus',
+   ...
+]
+
+MIDDLEWARE = [
+    'django_prometheus.middleware.PrometheusBeforeMiddleware',
+    # All your other middlewares go here, including the default
+    # middlewares like SessionMiddleware, CommonMiddleware,
+    # CsrfViewmiddleware, SecurityMiddleware, etc.
+    'django_prometheus.middleware.PrometheusAfterMiddleware',
+]
+```
+
+#### Modify urls.py
+Make sure you have this in the header: 
+```python
+from django.conf.urls import url
+
+Then add this under urlpatterns:
+urlpatterns = [
+    ...
+    url('', include('django_prometheus.urls')),
+]
+```
+
+#### Verify that metrics are being exported
+Restart the application and curl the /metrics endpoint:
+```bash
+python manage.py runserver
+curl localhost:8000/metrics
+```
+
+(Alternatively, visit [http://localhost:8000/metrics][localhost-metrics]
+from your web browser)
+
+You should see something like this:
+
+```
+# HELP python_gc_objects_collected_total Objects collected during gc
+# TYPE python_gc_objects_collected_total counter
+python_gc_objects_collected_total{generation="0"} 11716.0
+python_gc_objects_collected_total{generation="1"} 1699.0
+python_gc_objects_collected_total{generation="2"} 616.0
+# HELP python_gc_objects_uncollectable_total Uncollectable object found during GC
+# TYPE python_gc_objects_uncollectable_total counter
+python_gc_objects_uncollectable_total{generation="0"} 0.0
+python_gc_objects_uncollectable_total{generation="1"} 0.0
+python_gc_objects_uncollectable_total{generation="2"} 0.0
+# HELP python_gc_collections_total Number of times this generation was collected
+# TYPE python_gc_collections_total counter
+python_gc_collections_total{generation="0"} 7020.0
+python_gc_collections_total{generation="1"} 638.0
+python_gc_collections_total{generation="2"} 34.0
+# HELP python_info Python platform information
+# TYPE python_info gauge
+python_info{implementation="CPython",major="3",minor="8",patchlevel="0",version="3.8.0"} 1.0
+...
+```
+
+### Step 2 - Point Prometheus to your Django application metrics endpoint [](step2)
+
+*(Note: This section assumes that you have a locally running Prometheus
+instance.)*
+
+#### Update `prometheus.yml`. Under `scrape_configs:`, add:
+
+```yaml
+- job_name: django
+  scrape_interval: 10s
+  static_configs:
+  - targets:
+    - localhost:8000
+```
+
+>:TIP: Replace the job_name “django” with whatever you’d like as a prefix for
+your Django application metrics in Prometheus. For example, “webapp”, etc.)
+
+#### Restart Prometheus
+```bash
+./prometheus --config.file=prometheus.yml
+```
+
+#### Verify that Prometheus is scraping metrics from yourDjango application:
+Visit the [Prometheus Expression Browser][localhost-prom-browser] from your
```

Comment:
You should note that this is on the user's localhost, otherwise a reader may click on it and unknowingly be directed to a broken link (if it's not up and running)